python/src/aiconfig/default_parsers/dalle.py

```diff
@@ -0,0 +1,197 @@
+
+        Args:
+            prompt (str): The prompt to be serialized.
+            inference_settings (dict): Model-specific inference settings to be serialized.
+
+        Returns:
+            str: Serialized representation of the prompt and inference settings.
+        """
+        data = copy.deepcopy(data)
+
+        # assume data is completion params for HF text generation
+        prompt_input = data["prompt"]
+
+        # Prompt is handled, remove from data
+        data.pop("prompt", None)
+
+        model_metadata = ai_config.get_model_metadata(data, self.id())
+        prompt = Prompt(
+            name=prompt_name,
+            input=prompt_input,
+            metadata=PromptMetadata(model=model_metadata, parameters=parameters, **kwargs),
+        )
+        return [prompt]
+
+    # TODO (rossdanlm): Update documentation for args
+    async def deserialize(
+        self, prompt: Prompt, aiconfig: "AIConfigRuntime", _options, params: Optional[Dict] = {}
+    ) -> Dict:
+        """
+        Defines how to parse a prompt in the .aiconfig for a particular model
+        and constructs the completion params for that model.
+
+        Args:
+            Update this documentation... serialized_data (str): Serialized data from the .aiconfig.
+
+        Returns:
+            dict: Model-specific completion parameters.
+        """
+        # Get inputs from aiconfig
+        resolved_prompt = resolve_prompt(prompt, params, aiconfig)
+        model_settings = self.get_model_settings(prompt, aiconfig)
+
+        # Build Completion data
+        completion_data = refine_image_completion_params(model_settings)
+        completion_data["prompt"] = resolved_prompt
+        return completion_data
+
+    async def run_inference(self, prompt: Prompt, aiconfig, options, parameters) -> Output:
+        """
+        Invoked to run a prompt in the .aiconfig. This method should perform
+        the actual model inference based on the provided prompt and inference settings.
+
+        Args:
+            prompt (str): The input prompt.
+            inference_settings (dict): Model-specific inference settings.
+
+        Returns:
+            InferenceResponse: The response from the model.
+        """
+        completion_data = await self.deserialize(prompt, aiconfig, options, parameters)
+
+        # if stream enabled in runtime options and config, then stream. Otherwise don't stream.
+        stream = (options.stream if options else False) and (
+            not "stream" in completion_data or completion_data.get("stream") != False
+        )
+
+        print("Calling image generation. This can take several seconds, please hold on...")
+        response : ImagesResponse = self.client.images.generate(**completion_data)
+
+        outputs = []
+        if not stream:
+            # ImageResponse object also contains a "created" field for timestamp, should I store that somewhere?
+            # Ex: response=ImagesResponse(created=1700347843, data=[...])
```

Comment:
@LastMileNewGrad what do you think?

Reply:
One option is you can dump all the relevant data in metadata of the output

Reply:
Yea it's just that it seems a bit annoying to add this for every image output (since created field is the same for all of them), but I think I'll do it just to keep all the details. Will do this in future diff to keep commits small